lib/grape/validations/validators/default.rb

```diff
@@ -7,7 +7,24 @@ def initialize(attrs, options, required, scope)
       end
 
       def validate_param!(attr_name, params)
-        params[attr_name] = @default.is_a?(Proc) ? @default.call : @default unless params.key?(attr_name)
+        return if params.key? attr_name
+        params[attr_name] = if @default.is_a? Proc
+                              @default.call
+                            elsif @default.nil? ||
+                                  @default == true ||
+                                  @default == false ||
+                                  @default.is_a?(Symbol) ||
+                                  @default.is_a?(Numeric)
+                              # We know clone won't work in these cases, so just use the object.
+                              @default
+                            else
+                              # for the rest, try to clone but use the object if that fails.
+                              begin
+                                @default.clone
+                              rescue TypeError
```

Comment:
Also, `clone` retains `frozen`, I just want to ensure that this is what we want here. Maybe it should be `dup`?